src/redux/wallets.js

```diff
@@ -112,13 +124,46 @@ export const walletsSetSelected = wallet => async dispatch => {
   });
 };
 
-export const isCreatingAccount = val => dispatch => {
+export const setIsWalletLoading = val => dispatch => {
   dispatch({
     payload: val,
-    type: WALLETS_SET_IS_CREATING_ACCOUNT,
+    type: WALLETS_SET_IS_LOADING,
   });
 };
 
+export const setWalletBackedUp = (
+  walletId,
+  method,
+  backupFile = null
+) => async (dispatch, getState) => {
+  const { wallets, selected } = getState().wallets;
+  const newWallets = { ...wallets };
+  newWallets[walletId].backedUp = true;
+  newWallets[walletId].backupType = method;
+  if (backupFile) {
+    newWallets[walletId].backupFile = backupFile;
+  }
+  newWallets[walletId].backupDate = Date.now();
+  dispatch(walletsUpdate(newWallets));
+  if (selected.id === walletId) {
+    dispatch(walletsSetSelected(newWallets[walletId]));
+  }
+
+  // Reset the loading state 1 second later
+  setTimeout(() => {
+    dispatch(setIsWalletLoading(null));
+  }, 1000);
+
+  if (method === WalletBackupTypes.cloud) {
+    try {
+      await backupUserDataIntoCloud({ wallets: newWallets });
+    } catch (e) {
+      logger.error('SAVING WALLET USERDATA FAILED', e);
```

Comment:
Should we notify the user that the cloud backup actually didn't go through (right now I think it'll still show the user it was successful)

Reply:
The cloud backup and this are separate (this is only the public part of it) but you're right. We should treat it as an atomic operation